sbol2/validator.py

```diff
@@ -0,0 +1,172 @@
+            self.valid = False
+            self.digest_errors(output)
+        else:
+            self.digest_errors(output.strip(u"Validation successful, no errors."))
+            self.valid = True
+
+            if options.return_file:
+                with open(options.output_file, 'r') as file:
+                    self.result = file.read()
+
+    def broken_validation_request(self, command):
+        self.valid = False
+        self.errors = ["Something about your validation request is contradictory or poorly-formed!", " ".join(command)]
+
+    def json(self):
+        return self.__dict__
+
+
+class ValidationRun:
+    def __init__(self, options, validation_file, diff_file=None):
+        self.options = options
+        self.validation_file = validation_file
+        self.diff_file = diff_file
+
+    def execute(self):
+        result = ValidationResult(self.options.output_file, self.options.test_equality)
+
+        # Attempt to run command
+        jar_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "libSBOLj.jar")
+        command = self.options.command(jar_path, self.validation_file, self.diff_file)
+        try:
+            output = subprocess.check_output(command, universal_newlines=True, stderr=subprocess.STDOUT)
+            result.decipher(output, self.options)
+        except subprocess.CalledProcessError as exception:
+            # If the command fails, the file is not valid.
+            result.valid = False
+            result.errors += [exception.output, ]
+        except ValueError as ve:
+            print(traceback.print_tb(ve.__traceback__))
+            result.broken_validation_request(command)
+
+        return result.json()
+
+
+class ValidationOptions:
+    language = "SBOL2"
+    subset_uri = False
+    fail_on_first_error = False
+    provide_detailed_stack_trace = False
+    check_uri_compliance = True
+    check_completeness = True
+    check_best_practices = False
+    uri_prefix = False
+    version = False
+    insert_type = False
+    test_equality = False
+    return_file = True
+    main_file_name = "main file"
+    diff_file_name = "comparison file"
+
+    def __init__(self, return_file):
+        self.return_file = return_file
+
+    def build(self, work_dir, data):
+        for key, value in data.items():
+            setattr(self, key, value)
+        self.output_file = os.path.join(work_dir, str(uuid.uuid4()))
+
+        if self.language in ['SBOL1', 'SBOL2', 'SBML']:
+            self.output_file = self.output_file + ".rdf"
+        elif self.language == 'GFF3':
+            self.output_file = self.output_file + '.gff'
+        elif self.language == 'GenBank':
+            self.output_file = self.output_file + '.gb'
+        else:
+            self.output_file = self.output_file + '.fasta'
+
+    def command(self, jar_path, validation_file, diff_file=None):
+        command = ["/usr/bin/java", "-jar", jar_path, validation_file, "-o", self.output_file, "-l", self.language]
```

Comment:
`/usr/bin/java` seems platform specific. It's fine for now, but you may want to add a config option for the path to a JRE. Perhaps also mentioning minimum version of the JRE.